libraries/botbuilder-dialogs-adaptive-testing/src/setTestOptionsMiddleware.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * @module botbuilder-dialogs-adaptive-testing
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Middleware, TurnContext, ActivityTypes } from 'botbuilder-core';
+
+const CONVERSATION_STATE = 'ConversationState';
+
+export class SetTestOptionsMiddleware implements Middleware {
+    /**
+     * Processes an incoming event activity.
+     * @param context The context object for this turn.
+     * @param next The delegate to call to continue the bot middleware pipeline
+     */
+    public async onTurn(context: TurnContext, next: () => Promise<void>): Promise<void> {
```

Comment:
I meant that 'next' should be marked as an optional (?) parameter here. Otherwise, the null check below doesn't make sense.

Reply:
`onTurn` is defined in the interface "Middleware", and the next parameter is required.
But the required `next` is may still be assigned to `null`, for example:
```
const middleware = new SetTestOptionsMiddleware();
middleware.onTurn(context, null); //This is allowed by the typescript compiler
```

Reply:
It should not compile, [see this Typescript playground example](https://www.typescriptlang.org/play?#code/MYewdgzgLgBAZgVzMGBeGAKAhgLkwSjQD4YA3EASwBNDUSBvAKBhZiw3wG5GBfRxxMgxgEAG1FcgA).

The compiler settings must be too permissive at the moment, i.e., not using `strictNullChecks`. You should either change the signature to make `next` optional or remove the `if` check (and optionally throw a `TypeError` if next is not defined).

As time permits, we will invest in making our Typescript code more correct by enabling more strict compilation settings. There's no reason to write code now that doesn't conform to those stricter standards.

Reply:
It is strange that there is no error in my vscode. Hmmm.

Reply:
I think setting `strictNullChecks` (or, preferably, just `strict`) to `true` in the `tsconfig.json` file would do the trick.